### ISSUE TITLE
"Slow test card, declines after a few minutes" payment method never triggers purchaseErrorListener

### ISSUE THREAD

**Issue by simmotde** [dooboolab/react-native-iap#1226]
### Version of react-native-iap

5.1.3

### Version of react-native

0.61.2

### Platforms you faced the error (IOS or Android or both?)

Android

### Expected behavior

`purchaseErrorListener` will be called after a few minutes

### Actual behavior

`purchaseErrorListener` is not called even after > 1 hour

### Tested environment (Emulator? Real Device?)

Emulator (Nexus 5) and Real device (Huawei Nova 2i)

### Steps to reproduce the behavior

1. Purchase a consumable product (beta/sandbox mode)
2. Choose "Slow test card, declines after a few minutes" option from the payment methods
3. Complete purchase
4. Wait for error to occur (Error never happens here)

### Additional Info

I noticed that I can retrieve the products I purchased this way when I execute `getPurchaseHistory`. I tried calling `acknowledgePurchaseAndroid` and `consumePurchaseAndroid` with the `purchaseToken` I get from the response but it either returns `Error: Purchase failed with code 8` or `Google is indicating that we have some issue connecting to payment.` In the end I am forced to call `consumeAllItemsAndroid` so I can test again.

**Comment by simmotde** [comment]
I have just found out that when you try to purchase the same consumable product again within a certain period of time, Google Play will prompt that there is still a pending payment for the product. You have the option to complete the payment, in which case you will be redirected to Google Play Store app. If you cancel the pending purchase and come back to your app, `purchaseErrorListener` is called successfully.

But what if you never tried to repurchase the same consumable item? My assumption is that the pending purchase will **silently** expire and you won't have the option to cancel but you can then purchase the item again.

**Comment by hyochan** [comment]
@simmotde Could you summarize your concern here? Are you saying that the pending purchase should not expire?

**Comment by simmotde** [comment]
@hyochan No, I think the pending purchase should throw an error (trigger `purchaseErrorListener`) instead of expiring. I am not sure if this is an issue with `react-native-iap`, or this is how Google intended it to work for the purchase type. In any case, I will close this as I now know what happens and can handle it now.
